crates/oxc_linter/src/rules/no_duplicate_case.rs

```diff
@@ -0,0 +1,171 @@
+use oxc_ast::{AstKind, Span};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use rustc_hash::FxHashMap;
+
+use crate::{ast_util::calculate_hash, context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-duplicate-case): Disallow duplicate case labels")]
+#[diagnostic(severity(warning), help("Remove the duplicated case"))]
+struct NoDuplicateCaseDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoDuplicateCase;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow duplicate case labels
+    ///
+    /// ### Why is this bad?
+    ///
+    /// If a switch statement has duplicate test expressions in case clauses,
+    /// it is likely that a programmer copied a case clause but forgot to change the test expression.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// var a = 1;
+    /// switch (a) {
+    ///     case 1:
+    ///         break;
+    ///     case 1:
+    ///         break;
+    ///     case 2:
+    ///         break;
+    ///     default:
+    ///         break;
+    /// }
+    /// ```
+    NoDuplicateCase,
+    correctness
+);
+
+impl Rule for NoDuplicateCase {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::SwitchStatement(ss) = node.get().kind() {
+            let mut map = FxHashMap::default();
+            map.reserve(ss.cases.len());
+            for case in ss.cases.iter() {
+                if let Some(test) = case.test.as_ref() {
+                    let hash = calculate_hash(test);
+                    if map.insert(hash, case.span).is_some() {
+                        ctx.diagnostic(NoDuplicateCaseDiagnostic(case.span));
```

Comment:
should point to two `test.span()`

Reply:
fixed in [5e56d67](https://github.com/Boshen/oxc/pull/179/commits/5e56d673b80d96536217f15f9176f4c57c7af189)